CRM/Activity/ActionMapping.php

```diff
@@ -90,9 +90,9 @@ public function createQuery($schedule, $phase, $defaultParams) {
     $query['casContactTableAlias'] = NULL;
     $query['casDateField'] = 'e.activity_date_time';
 
-    if (!is_null($schedule->limit_to)) {
+    if (in_array($schedule->limit_to, [2, 3])) {
```

Comment:
probably need to do a non-hard-coded check based on CRM_Core_SelectValues::getLimitToValues() - or (better) a name check via the api

Reply:
hmm - on more looking I see an awful lot of hard-coded checks - made harder by the use of BAO.

I would proably use a helper function eg `BAO_ActionShedule::isLimitedTo()` - but I can also see that this PR would need a LOT  of change to get rid of the hard-coded numbers & I'm inclined to say  we should focus on getting it wrapped up rather than code nicities

Reply:
( I do think the `selectValues` one should be a 'proper' array though in the new format - that one will bite us later as the api increasingly cares)